services/datamanager/datasync/sync.go

```diff
@@ -81,23 +82,43 @@ func NewManager(identity string, client v1.DataSyncServiceClient, logger logging
 	cancelCtx, cancelFunc := context.WithCancel(context.Background())
 	logger.Debugf("Making new syncer with %d max threads", maxSyncThreads)
 	ret := syncer{
-		partID:             identity,
-		client:             client,
-		logger:             logger,
-		cancelCtx:          cancelCtx,
-		cancelFunc:         cancelFunc,
-		arbitraryFileTags:  []string{},
-		inProgress:         make(map[string]bool),
-		syncErrs:           make(chan error, 10),
-		syncRoutineTracker: make(chan struct{}, maxSyncThreads),
-		captureDir:         captureDir,
+		partID:            identity,
+		client:            client,
+		logger:            logger,
+		cancelCtx:         cancelCtx,
+		cancelFunc:        cancelFunc,
+		arbitraryFileTags: []string{},
+		inProgress:        make(map[string]bool),
+		syncErrs:          make(chan error, 10),
+		filesToSync:       make(chan string), //shold this be buffered?
```

Comment:
I'd set this to `maxSyncThreads`. But that would just be to smooth out some latency. Unbuffered is totally fine. The only risk is making the number too large.